evmcore/tx_pool.go

```diff
@@ -445,28 +437,9 @@ func (pool *TxPool) GasPrice() *big.Int {
 	pool.mu.RLock()
 	defer pool.mu.RUnlock()
 
-	return new(big.Int).Set(pool.gasPrice)
-}
-
-// SetGasPrice updates the minimum price required by the transaction pool for a
-// new transaction, and drops all transactions below this threshold.
-func (pool *TxPool) SetGasPrice(price *big.Int) {
-	pool.mu.Lock()
-	defer pool.mu.Unlock()
-
-	old := pool.gasPrice
-	pool.gasPrice = price
-	// if the min miner fee increased, remove transactions below the new threshold
-	if price.Cmp(old) > 0 {
-		// pool.priced is sorted by GasFeeCap, so we have to iterate through pool.all instead
-		drop := pool.all.RemotesBelowTip(price)
-		for _, tx := range drop {
-			pool.removeTx(tx.Hash(), true)
-		}
-		pool.priced.Removed(len(drop))
-	}
-
-	log.Info("Transaction pool price threshold updated", "price", price)
+	return new(big.Int).Set(
```

Comment:
The `GasPrice` function seems to be nowhere used any more. The only reference is in the `gossip.TxPool` interface, but there it can be deleted as well as it is not used anywhere.

Reply:
It is required by the interface to the gossip protocol:  https://github.com/Fantom-foundation/Sonic/blob/63190cb72ba47ca67ce8162046ab0b0cc18a76de/gossip/protocol.go#L113